lib/modules/datasource/hermit/index.ts

```diff
@@ -0,0 +1,146 @@
+import { logger } from '../../../logger';
+import { cache } from '../../../util/cache/package/decorator';
+import { GithubHttp } from '../../../util/http/github';
+import { regEx } from '../../../util/regex';
+import { streamToString } from '../../../util/streams';
+import { id } from '../../versioning/hermit';
+import { Datasource } from '../datasource';
+import { GithubReleasesDatasource } from '../github-releases';
+import { getApiBaseUrl } from '../github-releases/common';
+import type {
+  GithubRelease,
+  GithubReleaseAsset,
+} from '../github-releases/types';
+import type { GetReleasesConfig, ReleaseResult } from '../types';
+import type { HermitSearchResult } from './types';
+
+/**
+ * Hermit Datasource searches a given package from the specified `hermit-pakcages`
+ * repository. It expects the search manifest to come from an asset `index.json` from
+ * a release named index. Any error fetching, parsing and finding the package from the given
+ * registryUrl will be thrown
+ */
+export class HermitDatasource extends Datasource {
+  static readonly id = 'hermit';
+
+  override readonly customRegistrySupport = true;
+
+  override readonly registryStrategy = 'first';
+
+  override readonly defaultVersioning = id;
+
+  override readonly defaultRegistryUrls = [
+    'https://github.com/cashapp/hermit-packages',
+  ];
+
+  githubHttp: GithubHttp;
```

Comment:
```suggestion
  http: GithubHttp;
```
override base property